### ISSUE TITLE
False negative for unused return

### ISSUE THREAD

**Issue by leifg** [rrrene/credo#527]
### Environment

* Credo version (`mix credo -v`): 0.9.1
* Erlang/Elixir version (`elixir -v`):
* Operating system:

### What were you trying to do?

`mix credo --strict` on the following file (no .credo.exs in project)

```elixir
defmodule CredoExample do
  @moduledoc """
  Simple Example
  """

  def error do
    module().call(Path.join("~", "foo"))

    "something"
  end

  def no_error(:one) do
    call(Path.join("~", "foo"))

    "something"
  end

  def no_error(:two) do
    module().call(Path.join("~", "foo"))
  end

  defp call(path) do
    IO.puts path
  end

  defp module do
    CredoExample
  end
end

```

### Expected outcome

No issues reported

### Actual outcome

Line 7 (the body of the function `error`) raises the warning:

```
  Warnings - please take a look
┃
┃ [W] ↗ There should be no unused return values for Path.join().
┃       lib/credo_example.ex:7:19 #(CredoExample.error)
```

In 0.9.0 the behavior is fine.

**Comment by dotboris** [comment]
Looks like this was fixed in https://github.com/rrrene/credo/commit/517699f82fc6ebe266152d997b64878b334e7bd8

**Comment by dotboris** [comment]
The fix is also part of v0.9.2
